src/lib/Libifl/pbsD_submit.c

```diff
@@ -219,6 +219,22 @@ __pbs_submit(int c, struct attropl  *attrib, char *script, char *destination, ch
 			extend = EXTEND_OPT_IMPLICIT_COMMIT;
 	}	
 
+	if (destination != NULL && msvr_mode()) {
+		/* 
+		 * Reached here means job is submitted to non default queue
+		 * Also check if it is a reservation queue
+		 */
+		if (destination[0] == 'R' || destination[0] == 'S' || destination[0] == 'M') {
+			int start = 0;
```

Comment:
this is not a good check, any queue can start with 'R', 'S' or 'M', not just reservation queues, right? You don't really need this check, just pass destination as it is to get_job_resv_location_hint(), if it returns -1, just proceed normally, I'm guessing the job will be submitted to a random server. get_job_resv_location_hint() should not fail if the queue name is a valid reservation queue.

Reply:
Also, what will happen to existing reservations when a site switches from single to multi-server? How will this code work with old reservation ids? (which might be owned by any one of the N servers)

Reply:
> this is not a good check, any queue can start with 'R', 'S' or 'M', not just reservation queues, right? You don't really need this check, just pass destination as it is to get_job_resv_location_hint(), if it returns -1, just proceed normally, I'm guessing the job will be submitted to a random server. get_job_resv_location_hint() should not fail if the queue name is a valid reservation queue.

Please look into my latest commit. I have refactored the places where we check if it is resv or not.  Since we removed the above check now even if somebody submits a job to non-default queue we still call get_shard_obj_location_hint(). As this is not a very costly function with respect to performance I feel it is ok and that we are doing this in the client side not on the server side.

Reply:
> Also, what will happen to existing reservations when a site switches from single to multi-server? How will this code work with old reservation ids? (which might be owned by any one of the N servers)

I think a similar case arises for jobs as well. When we come up with an approach to handle upgrades this will be taken care of. I will make a note on it.